consumption.tex

```diff
@@ -0,0 +1,89 @@
+
+  \begin{itemize}
+    \item $\VerTxIncl(b, tx, \tau)$: Given block $b$, transaction $tx$ and a proof $\tau$, return $\true$ if $\tau$ is a valid proof for the fact that $tx \in b$.
+    \item $\TxHasTransfer(tx, addr, amount)$: Given transaction $tx$, an address $addr$ and an amount, return $\true$ if $tx$ contains a value transfer of the specified amount towards the specified address.
+  \end{itemize}
+\end{definition}
+
+More concretely, in Bitcoin, each block header contains a commitment to the set of transaction ids in the block in the form of a Merkle Tree root. Thus, for Bitcoin the proof $\tau$ for $\VerTxIncl$ is a Merkle Tree inclusion proof. Ethereum stores a similar commitment in its header, in the form of the root of a Merkle-Patricia Trie~\cite{wood2014ethereum}.
+
+For a verifier to be convinced that a burn has taken place on a source blockchain, they need to ensure that the burn is contained in a transaction which belongs to the source blockchain. Specifically, the following data need to be supplied to the smart contract:
+
+\begin{itemize}
+  \item \textsf{tx}: The transaction which contains the burn on the source blockchain.
+  \item \textsf{blk}: The block where the aforementioned transaction is contained.
+  \item \textsf{txInclusion}: An inclusion proof showing that $\mathsf{tx} \in \mathsf{blk}$ such that $\VerTxIncl(\textsf{blk}, \textsf{tx}, \textsf{txInclusion}) = \true$.
+  \item \textsf{blockConnection}: A proof that \textsf{blk} is contained in the best source blockchain.
+\end{itemize}
+
+Putting these together, the final proof is the tuple $\beta = \sf(tx, txInclusion, blk, blockConnection)$.
+
+\subsection{Proof-of-Work Sidechains}
+Transferring value from a source to a target blockchain has already been a subject of research. A mechanism for the so-called 1-way pegs is described in~\cite{pow-sidechains}. A 1-way peg is established when there is the ability to transfer an amount of source cryptocurrency to a target blockchain in a trustless manner, as we wish to do. Generally, the prover wishes to convince the verifier that an event has taken place on the source blockchain. In that work, an event describes a Solidity event emitted by a smart contract on some chain. It is described as a triplet $e = (chain, contract, aux)$, where $chain$ is an identifier for the chain, $contract$ refers to the address of the contract and $aux$ contains the parameters of the event emitted.
+
+The contract \textsf{crosschain} is defined, which is initialized with the genesis block of the other blockchain $\mathcal{G}$. This contract is responsible for receiving, verifying and then storing events on the other blockchain. It relies on a function $\mathsf{verify}^{e,\mathcal{G}}(\mathcal{P})$, where $\mathcal{P}$ is a set of proofs for the event, which returns $\true$ if and only if the event is valid based on the set of proofs provided.
+
```

Comment:
Some text describing what the list will be about is appropriate here. Perhaps "The \textsc{crosschain} smart contract supplies the following methods:"